optaplanner-examples/src/test/java/org/optaplanner/examples/machinereassignment/solver/score/MachineReassignmentConstraintProviderTest.java

```diff
@@ -0,0 +1,384 @@
+
+        constraintVerifier.verifyThat(MachineReassignmentConstraintProvider::processMoveCost)
+                .given(globalPenaltyInfo, machine1, machine2, process, processAssignment, processAssignment2,
+                        processAssignment3)
+                .penalizesBy(20L);
+    }
+
+    @Test
+    public void serviceMoveCost() {
+        MrGlobalPenaltyInfo globalPenaltyInfo = new MrGlobalPenaltyInfo();
+        globalPenaltyInfo.setServiceMoveCostWeight(10);
+
+        MrMachine machine1 = new MrMachine();
+        MrMachine machine2 = new MrMachine();
+
+        MrService service1 = new MrService(1L);
+        MrService service2 = new MrService(2L);
+        // service2 has only one process moving, while service1 has two processes moving => wins
+        MrProcess process1 = new MrProcess(service1);
+        MrProcess process2 = new MrProcess(service1);
+        MrProcess process3 = new MrProcess(service2);
+
+        MrProcessAssignment processAssignment1 = new MrProcessAssignment(0L, process1, machine1, machine2);
+        MrProcessAssignment processAssignment2 = new MrProcessAssignment(1L, process2, machine1, machine2);
+        MrProcessAssignment processAssignment3 = new MrProcessAssignment(1L, process3, machine1, machine2);
+
+        constraintVerifier.verifyThat(MachineReassignmentConstraintProvider::serviceMoveCost)
+                .given(globalPenaltyInfo, machine1, machine2, service1, service2, process1, process2, process3,
+                        processAssignment1, processAssignment2, processAssignment3)
+                .penalizesBy(2 * 10L);
+    }
+
+    @Test
+    public void machineMoveCost() {
+        MrGlobalPenaltyInfo globalPenaltyInfo = new MrGlobalPenaltyInfo();
+        globalPenaltyInfo.setMachineMoveCostWeight(10);
+
+        MrMachine machine1 = new MrMachine();
+        MrMachine machine2 = new MrMachine();
+
+        Map<MrMachine, Integer> costMapFromMachine1 = new HashedMap<>();
+        costMapFromMachine1.put(machine2, 20);
+        machine1.setMachineMoveCostMap(costMapFromMachine1);
+
+        Map<MrMachine, Integer> costMapFromMachine2 = new HashedMap<>();
+        costMapFromMachine2.put(machine1, 0);
+        machine2.setMachineMoveCostMap(costMapFromMachine2);
+
+        MrProcess process1 = new MrProcess();
+        MrProcess process2 = new MrProcess();
+        MrProcess process3 = new MrProcess();
+
+        MrProcessAssignment processAssignment1 = new MrProcessAssignment(0L, process1, machine1, machine2);
+        MrProcessAssignment processAssignment2 = new MrProcessAssignment(0L, process2, machine1, machine2);
+        MrProcessAssignment processAssignment3 = new MrProcessAssignment(0L, process3, machine2, machine1);
+
+        /*
+         * 2 processes are moving from machine1 to machine2, which has a cost of 20 => 2 * 20 * 10 (global penalty) = 400.
+         * The process3 moves from machine2 to machine1, which has a zero cost.
+         */
```

Comment:
code style suggestion: this takes up 4 lines for 2 lines of content, typically we just use two `//` for 2 lines of non-javadoc comments.